### PR TITLE
Enhancement(CX-1698): add link taps action and screen auctionResultsForArtistYouFollow to cohesion

### PR DESCRIPTION
**Pull request by lordkiz**
The type of this PR is: **Enhancement**

This PR resolves [CX-1800]

### Description
- Adds a general ActionType for taps on inline Hypertext/links

- Adds OwnerType.auctionResultsForYou to ScreenOwnerTyper to allow for tracking of ActionResultsForYou screen

### PR Checklist (tick all before merging)

- [x] If I've added a new file to the tree I've exported it from the common `index.ts`
- [x] I've added comments with examples for any new interfaces or helpers and ensured that they're in the docs 
- [x] No platform-specific terminology has been used outside of `click` and `tap` (platform is inferred by the DB storing events)

[CX-1800]: https://artsyproduct.atlassian.net/browse/CX-1800

### CODE REVIEW

File: src/Schema/Events/Tap.ts [hunk 1]
```diff
@@ -647,3 +647,31 @@ export interface TappedVerifyIdentity {
...
+ */
+export interface TappedLink {
+  action: ActionType.tappedLink
+  context_module: ContextModule
+  context_screen_owner_type: ScreenOwnerType
+  context_screen_owner_id?: string
+  context_screen_owner_slug?: string
+  destination_screen_owner_type?: ScreenOwnerType
+  destination_screen_owner_id?: string
+  destination_screen_owner_slug: string
```
Comment (mikehrom) [inline src/Schema/Events/Tap.ts:675]: i think maybe `destination_screen_owner_slug` is optional too, unless you're referring to `destination_path` -- which I believe we use as a property in other events. `destination_path` works for internal or external links, where external links won't have a `slug` that corresponds to some artsy data object (like an artwork, artist, or partner).

Comment (mikehrom) [reply src/Schema/Events/Tap.ts:675]: tl;dr:
- suggesting `destination_path` as required prop
- suggesting `destination_screen_owner_slug?`

**Review by mikehrom — COMMENTED** [review]
thanks for this! i commented in line with one suggestion. we can sync on slack or zoom if you need, too.

File: src/Schema/Events/Tap.ts (hunk 1 above)
Comment (mikehrom) [reply src/Schema/Events/Tap.ts:675]: Correct, I'd agree with that 👍

Comment (lordkiz) [reply src/Schema/Events/Tap.ts:675]: Ok thanks

**Comment by artsyit** [comment]
:rocket: PR was released in [`v2.19.1`](https://github.com/artsy/cohesion/releases/tag/v2.19.1) :rocket:

**Pull request merged**